File: models/AuthUser.js
const {DataTypes} = require('sequelize');
const {sequelize} = require('../config/database');

const AuthUser = sequelize.define('AuthUser', {
    user_id: {
        type: DataTypes.BIGINT,          // removed .UNSIGNED
        autoIncrement: true,
        primaryKey: true
    },
    email: {
        type: DataTypes.STRING(255),
        unique: true,
        allowNull: false
    },
    password_hash: {
        type: DataTypes.STRING(255),
        allowNull: false
    },
    full_name: {
        type: DataTypes.STRING(120),
        allowNull: false
    },
    profile_photo: {
        // NVARCHAR(MAX) on MSSQL (good for base64 strings). If you store raw bytes, use BLOB/VARBINARY via DataTypes.BLOB.
        type: DataTypes.TEXT,
        allowNull: true
    },
    is_active: {
        type: DataTypes.BOOLEAN,          // maps to BIT
        defaultValue: true
    },
    email_verified: {
        type: DataTypes.BOOLEAN,          // maps to BIT
        defaultValue: false
    },
    created_at: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW
    },
    updated_at: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW
    },
    last_active: {
        type: DataTypes.DATE,
        allowNull: true
    }
}, {
    tableName: 'users',
    schema: 'dbo',                      // <- optional but recommended on MSSQL
    timestamps: false,
    hooks: {
        // keep updated_at fresh (MSSQL has no ON UPDATE CURRENT_TIMESTAMP)
        beforeUpdate: (instance) => {
            instance.set('updated_at', new Date());
        }
    }
});

module.exports = AuthUser;

File: models/auth_user.py
# models/auth_user.py
from __future__ import annotations
from typing import Optional
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import BigInteger, String, Boolean, DateTime, Integer
from datetime import datetime
from .base import Base


class AuthUser(Base):
    __tablename__ = "users"
    __table_args__ = {"schema": "dbo"}

    user_id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    email: Mapped[str] = mapped_column(String(255), unique=True, nullable=False)
    # DB column is named 'password'
    password_hash: Mapped[str] = mapped_column("password", String(255), nullable=False)

    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    last_active: Mapped[Optional[datetime]] = mapped_column(DateTime, nullable=True)

    # Store the role_id as an integer; DB enforces FK, ORM does not.
    user_role_id: Mapped[int] = mapped_column(Integer, nullable=False)

    # Explicit, view-only relationship to Role without FK constraints in ORM
    Role = relationship(
        "Role",
        primaryjoin="AuthUser.user_role_id == Role.role_id",
        uselist=False,
        viewonly=True,
    )

    # One-to-one with Employee (no ORM FK on Employee side either)
    Employee = relationship("Employee", uselist=False, back_populates="User")
